packages/lib/src/check.ts

```diff
@@ -20,10 +21,28 @@ const check = async (
     const compacted = await jsonld.compact(expanded, jsonldDoc['@context'], {
       documentLoader,
     });
+    const delimiter = '.';
     // Check which keys have been removed
-    const keys = Object.keys(jsonldDoc).filter(isNotJsonLdPropery);
-    const newKeysSet = new Set(Object.keys(compacted));
-    const difference = keys.filter(key => !newKeysSet.has(key));
+    let keys = Object.keys(flatten(jsonldDoc, { delimiter }))
```

Comment:
I think this approach will have problems if there are nested properties that contain `.` in the name. Could you add a test that covers that case?

Reply:
I've done something similar in tech spikes, having the dot is problematic if you are needing to unflatten to go back to the original object, and before flattening the key had a dot in it. When unflattening it will be treated as a nested object which isn't correct.  For what we are doing it shouldn't be a problem because we are only flattening and not unflattening the object.  But I could use a different delimiter/separator when flattening, I used `|` in the tech spike project.  Let me know what you think.

Reply:
I added a test case for document that has dot in key, works as expected with iri mapped or not.